### DISCUSSION TITLE
Workbench beta phase 2022-11-07 to 2023-02-13

### DISCUSSION THREAD

**Discussion by zkamvar** [datacarpentry/OpenRefine-ecology-lesson#275] [Announcements]
Hello @datacarpentry/openrefine-ecology-lesson-maintainers,

Thank you for your patience during the last quarter. 
I am pleased to announce that the **transition to beta phase for this lesson will take place in 5 weeks on 2022-11-07**.

## Timing

the beta phase is divided into the following schedule of stages

 - 2022-11-07: pre-beta stage (two repos, two sites)
 - 2022-12-05: beta stage (one repo, two sites)
 - 2023-02-13: pre-release stage (one repo, one site) 

Please let me know if this does not work for you and I will do my best to accommodate. 
Importantly: **please resolve (close or merge) any open pull requests before 2022-12-05** because the beta stage will involve changing the folder structure of this repository which will render any existing pull requests invalid.

## Pre Beta Tasks

During this first pre-beta stage of the Beta Phase, I will give you access to https://github.com/fishtree-attempt/OpenRefine-ecology-lesson and encourage you to test out the following things on the lesson to get familiar with The Workbench. Here are some suggestions:

 - clone the lesson and build it locally
 - (optionally trigger and) merge and automated workflow update PRs that @carpentries-bot opens
 - review a PR that includes some failing accessibility checks
 - (R lessons only) review a PR using the preview of rendered changes
 - fix a simple typo in a lesson - perhaps in a relatively non-obvious location e.g. in the setup instructions
 - write a link from one episode to another in the same lesson 
 
I will be recording a short video describing the changes you will see in the next week, so be sure to keep your eyes out for that.

**Comment by zkamvar** [comment]
Hi @datacarpentry/openrefine-ecology-lesson-maintainers,

I have come to realise that, while I got the initial okay to pilot this lesson for the workbench when I made the call earlier this year, I never got a second confirmation on that decision from the maintainers.

At the moment, your lesson is set to enter the beta phase on Monday, but I would like to make sure we have a consensus among maintainers to enter this testing phase. The dates are not set in stone and because this is a beta, I want to make sure this is not a surprise to anyone. 

**Please reply to confirm or reject your consent to enter the beta phase for The Carpentries Workbench**

**Reply by villanueval** [reply]
Hey @zkamvar, the recent PR got a bit out of hand and will make it more complicated, so I don't think the timeline will work for this lesson. 

If we can get the lesson back on track to be useful for the tests, I'll let you know.

**Reply by zkamvar** [reply]
Thank you for the reply @villanueval. I will put this lesson outside of the beta phase for now and we will revisit later.

**Comment by zkamvar** [comment]
## Workbench Beta Phase Timeframe Adjustment

> Note: this is identical to the message I sent directly via email

The beta phase has been in progress since the end of October, but **because of various changing circumstances, I will be pushing back the beta phase timeline according to the revised schedule below**, synchronising all of the beta phase repositories. With this new timeline, you will have more time to explore the repository snapshots, I will have time to provide more clear communications, and—because everyone will go through each stage synchronously—you will all be able to exchange tips and tricks together. 

Note that Toby Hodges, Vini Salazar, and I will be hosting weekly helpdesk sessions, whose times we will send to you soon. 

Here is the revised schedule, which you can also find [on The Workbench beta phase website](https://carpentries.github.io/workbench/beta-phase.html):

 - [2022-12-12 (pre-beta)](https://carpentries.github.io/workbench/beta-phase.html#pre-beta): (re-)release all beta phase lessons to the pre-beta stage, which will have a parallel snapshot at https://github.com/fishtree-attempt/ hosting the Workbench preview on https://preview.carpentries.org/. 
    - During this time, we encourage you to explore the repository. Here is a [list of tasks to get started](https://carpentries.github.io/workbench/beta-phase.html#pre-beta-tasks). Remember, This preview version is temporary, so you do not have to worry about making any mistakes here. 
    - Take the time to merge or resolve any open PRs. And yes, this is a big task and it is helpful to [remember how to say 'no'](https://opensource.guide/best-practices/#learning-to-say-no) in a way that respects the contributor’s time and effort. If you feel like you need help with a particular PR, please contact Zhian and he will help craft a response.
 - [2023-02-06 (beta)](https://carpentries.github.io/workbench/beta-phase.html#beta): do in-place conversion of all beta phase lessons to use The Workbench by default. The default lesson site will still point to a snapshot of the jekyll version from 2023-02-06, and the preview lesson site will show the Workbench hosted on the main branch of the repository. 
    - IMPORTANT: before this happens, all maintainers MUST be aware of this change and must merge or resolve any open PRs. Remember that it’s okay to say no and/or contact Zhian for wordsmithing.
    - Zhian will create a video describing what this process is doing
    - The broader maintainer community will be alerted that they have three months to merge or resolve their open PRs at this time.
 - [2023-04-03 (pre-release)](https://carpentries.github.io/workbench/beta-phase.html#pre-release): The Workbench will become the default version of the lesson and the Jekyll version will be removed. A blog post will go out alerting everyone of the change. 


**We will make the conversion of all of our lessons to use The Workbench on 2023-05-01**


Thank you once again for agreeing to participate in the beta phase of The Carpentries Workbench. Your efforts in testing the infrastructure on your established lessons are invaluable to improving the workbench and I want to make sure that we take the time to get that right.

**Reply by zkamvar** [reply]
Note: given our discussion above, I do not have any expectations of this lesson returning to the beta phase, but I wanted to keep you all in the loop just in case.

**Comment by zkamvar** [comment]
Please disregard my previous (deleted) comment.
